02_jwt_auth_flow/app.js

```diff
@@ -16,6 +16,9 @@ app.use(express.json());
 app.set("view engine", "ejs");
 app.set("views", path.join(__dirname, "views/pages"));
 
+// app.use(express.static(`${__dirname}/public`));
+
+
 // Routes
```

Comment:
this comment may be redundant by just saying "routes", Perhaps you want to provide additional context
```suggestion
/**
* the routes mount point"
* all the view layouts are mounted at / path 
* and the controllers are mounted at /api/auth
* @see /routes for more details 
```

Reply:
See https://jsdoc.app/ for best practices on documenting JavaScript code. Finally commenting should be approached with the idea of explaining one's code to someone, I added some examples below

```js 

import fs from "node:fs/promises";
import { build } from "./build.js";
import { EXERCISE_V2_SOURCE_DIRECTORY } from "./index.js";
import { log_error_message } from "./utils.js";

/**
 * @function build_all_exercises - compiles asynchronously for production mode 
 *
 * @params void
 *
 * @returns void
 */

export async function build_all_exercises() {
  const exercises = await fs
    .readdir(EXERCISE_V2_SOURCE_DIRECTORY)
    .catch((error) => {
      log_error_message("Build failed due to " + error.message);
    });
  for (const entry of exercises) {
    /**
     * ignore all "dotfiles" found in the directory 
     * this could be files like .git, .env or .DS_store for Apple laptops 
     */
    if (!entry.startsWith(".")) {
      build(entry);
    }
  }
}
```